mlflow/models/docker_utils.py

```diff
@@ -91,16 +126,24 @@ def _build_image(image_name, entrypoint, mlflow_home=None, custom_setup_steps_ho
     :param custom_setup_steps_hook: (Optional) Single-argument function that takes the string path
            of a dockerfile context directory and returns a string containing Dockerfile commands to
            run during the image build step.
+    :param python_only: To build docker image for python flavor only.
     """
     mlflow_home = os.path.abspath(mlflow_home) if mlflow_home else None
     with TempDir() as tmp:
         cwd = tmp.path()
-        install_mlflow = _get_mlflow_install_step(cwd, mlflow_home)
-        custom_setup_steps = custom_setup_steps_hook(cwd) if custom_setup_steps_hook else ""
-        with open(os.path.join(cwd, "Dockerfile"), "w") as f:
-            f.write(_DOCKERFILE_TEMPLATE.format(
-                install_mlflow=install_mlflow, custom_setup_steps=custom_setup_steps,
-                entrypoint=entrypoint))
+        install_mlflow = _get_mlflow_install_step(cwd, mlflow_home, python_only)
+        custom_setup_steps = custom_setup_steps_hook(cwd) \
+            if custom_setup_steps_hook else ""
+        if python_only:
+            with open(os.path.join(cwd, "Dockerfile"), "w") as f:
+                f.write(_DOCKERFILE_PYTHON_TEMPLATE.format(
+                    install_mlflow=install_mlflow, custom_setup_steps=custom_setup_steps,
+                    entrypoint=entrypoint))
+        else:
+            with open(os.path.join(cwd, "Dockerfile"), "w") as f:
+                f.write(_DOCKERFILE_TEMPLATE.format(
+                    install_mlflow=install_mlflow, custom_setup_steps=custom_setup_steps,
+                    entrypoint=entrypoint))
```

Comment:
No need for the duplication.

```suggestion
        dockerfile_template = _DOCKERFILE_PYTHON_TEMPLATE if python_only else _DOCKERFILE_TEMPLATE
        with open(os.path.join(cwd, "Dockerfile"), "w") as f:
            f.write(
                dockerfile_template.format(
                    install_mlflow=install_mlflow,
                    custom_setup_steps=custom_setup_steps,
                    entrypoint=entrypoint,
                )
            )
```

Reply:
Thank you for the feedback. Made the suggested code changes and testing is in progress.

Reply:
Testing completed.